fileHub/js-app/app/services/api-service/api-service.js

```diff
@@ -52,30 +56,32 @@ export class ApiService {
 
     const responseBody = await response.json();
 
-    if (response.ok) {
-      return responseBody;
-    } else if (response.status === 422) {
+    if (response.status === 422) {
       const errors = responseBody.map((responseError) =>
         new ValidationErrorCase(responseError.field, responseError.message));
       throw new UnprocessableEntityError(errors);
-    } else if ((response.status >= 400 && response.status <= 421) ||
-				(response.status >= 423 && response.status < 500)) {
+    }
+
+    if ((response.status >= 400 && response.status < 500)) {
       throw new ClientServerError(responseBody.message);
-    } else if (response.status === 500) {
-      throw new ServerError(responseBody.message);
     }
```

Comment:
This code is duplicated in both methods so can be moved to a private reusable method.